include/swift/Runtime/RuntimeFunctions.def

```diff
@@ -810,13 +810,15 @@ FUNCTION(UpdateClassMetadata,
 //                                             ClassLayoutFlags flags,
 //                                             size_t numFields,
 //                                             TypeLayout * const *fieldTypes,
-//                                             size_t *fieldOffsets);
+//                                             size_t *fieldOffsets,
+//                                             objc_class_stub *classStub);
 FUNCTION(InitClassMetadata2,
          swift_initClassMetadata2, SwiftCC,
          RETURNS(TypeMetadataDependencyTy),
          ARGS(TypeMetadataPtrTy, SizeTy, SizeTy,
               Int8PtrPtrTy->getPointerTo(),
-              SizeTy->getPointerTo()),
+              SizeTy->getPointerTo(),
+              ObjCResilientClassStubTy->getPointerTo()),
```

Comment:
Sorry, what makes this not an ABI-breaking change?

Reply:
Adding a new function parameter is not ABI breaking (as long as you don't depend on its value except when you know the caller passed it), but this whole part of the change is going away and instead the stub will be stored in the class context descriptor.